salt/_logging/impl.py

```diff
@@ -0,0 +1,1438 @@
+SORTED_LEVEL_NAMES = [
+    l[0] for l in sorted(six.iteritems(LOG_LEVELS), key=lambda x: x[1])
+]
+
+MODNAME_PATTERN = re.compile(r'(?P<name>%%\(name\)(?:\-(?P<digits>[\d]+))?s)')
+
+
+class __NullLoggingHandler(logging.NullHandler):
+    '''
+    This class exists just to better identify which temporary logging
+    handler is being used for what.
+    '''
+
+
+class SaltLogRecord(logging.LogRecord):
+    def __init__(self, *args, **kwargs):
+        logging.LogRecord.__init__(self, *args, **kwargs)
+        self.bracketname = '[{:<17}]'.format(self.name)
+        self.bracketlevel = '[{:<8}]'.format(self.levelname)
+        self.bracketprocess = '[{:>5}]'.format(self.process)
+
+
+class SaltColorLogRecord(SaltLogRecord):
+    def __init__(self, *args, **kwargs):
+        SaltLogRecord.__init__(self, *args, **kwargs)
+
+        reset = TextFormat('reset')
+        clevel = LOG_COLORS['levels'].get(self.levelname, reset)
+        cmsg = LOG_COLORS['msgs'].get(self.levelname, reset)
+
+        self.colorname = '{}[{:<17}]{}'.format(LOG_COLORS['name'],
+                                               self.name,
+                                               reset)
+        self.colorlevel = '{}[{:<8}]{}'.format(clevel,
+                                               self.levelname,
+                                               reset)
+        self.colorprocess = '{}[{:>5}]{}'.format(LOG_COLORS['process'],
+                                                 self.process,
+                                                 reset)
+        self.colormsg = '{}{}{}'.format(cmsg, self.getMessage(), reset)
+
+
+# Store an instance of the current logging logger class
+LOGGING_LOGGER_CLASS = logging.getLoggerClass()
+
+
+def set_log_record_factory(factory):
+    '''
+    Set the logging  log record factory
+    '''
+    get_log_record_factory.__factory__ = factory
+    if not six.PY2:
+        logging.setLogRecordFactory(factory)
+
+
+def get_log_record_factory():
+    '''
+    Get the logging  log record factory
+    '''
+    try:
+        return get_log_record_factory.__factory__
+    except AttributeError:
+        set_log_record_factory(SaltLogRecord)
+        return get_log_record_factory.__factory__
+
+
+class SaltLoggingClass(six.with_metaclass(LoggingMixinMeta, LOGGING_LOGGER_CLASS, NewStyleClassMixin)):
+    def __new__(cls, *args):
+        '''
+        We override `__new__` in our logging logger class in order to provide
+        some additional features like expand the module name padding if length
+        is being used, and also some Unicode fixes.
+
+        This code overhead will only be executed when the class is
+        instantiated, i.e.:
+
+            logging.getLogger(__name__)
+
+        '''
+        instance = super(SaltLoggingClass, cls).__new__(cls)
+
+        try:
+            max_logger_length = len(max(
+                list(logging.Logger.manager.loggerDict), key=len
+            ))
```

Comment:
That was carried over from the previous implementation, I'm sure there was a reason, but I'll investigate.

Reply:
Consider the following code:
```python
        import timeit
        t1_max_logger_length = len(max(
            list(logging.Logger.manager.loggerDict), key=len
        ))
        t1 = timeit.timeit('ParserBase.t1()', setup='from tests.unit.utils.test_parsers import ParserBase')
        t2_max_logger_length = max(
            len(name) for name in logging.Logger.manager.loggerDict
        )
        t2 = timeit.timeit('ParserBase.t2()', setup='from tests.unit.utils.test_parsers import ParserBase')
        print('t1', t1_max_logger_length, t1)
        print('t2', t2_max_logger_length, t2)
        t1_max_logger_length = len(max(
            list(logging.Logger.manager.loggerDict), key=len
        ))
        t1 = timeit.timeit('ParserBase.t1()', setup='from tests.unit.utils.test_parsers import ParserBase')
        t2_max_logger_length = max(
            len(name) for name in logging.Logger.manager.loggerDict
        )
        t2 = timeit.timeit('ParserBase.t2()', setup='from tests.unit.utils.test_parsers import ParserBase')
        print('t1', t1_max_logger_length, t1)
        print('t2', t2_max_logger_length, t2)


    @staticmethod
    def t1():
        max_logger_length = len(max(
            list(logging.Logger.manager.loggerDict), key=len
        ))

    @staticmethod
    def t2():
        max_logger_length = max(
            len(name) for name in logging.Logger.manager.loggerDict
        )
```
Now, the outcome of that:
```
t1 27 8.73357391357                                                                                             
t2 29 12.3908040524                                                                                                                                                                                                              
t1 29 8.77210521698                                                                                                                                                                                                              
t2 29 12.4122331142
```
On the first iteration, the max length wasn't equal, maybe another logger was added, so let's ignore that on the first run, specially since on the second `timeit` they both report the same length.
Now, the existing implementation is tad faster than your proposed solution.
Maybe I reached this same conclusion the first time around.